Std/Data/Sum/Lemmas.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Yury G. Kudryashov
+-/
+
+import Std.Data.Sum.Basic
+import Std.Tactic.Ext
+
+/-!
+# Disjoint union of types
+
+Theorems about the definitions introduced in `Std.Data.Sum.Defs`.
+-/
+
+namespace Sum
+
+@[simp] theorem «forall» {p : Sum α β → Prop} : (∀ x, p x) ↔ (∀ a, p (inl a)) ∧ ∀ b, p (inr b) :=
+  ⟨fun h => ⟨fun _ => h _, fun _ => h _⟩, fun ⟨h₁, h₂⟩ => Sum.rec h₁ h₂⟩
+
+@[simp] theorem «exists» {p : Sum α β → Prop} : (∃ x, p x) ↔ (∃ a, p (inl a)) ∨ ∃ b, p (inr b) :=
```

Comment:
```suggestion
@[simp] protected theorem «forall» {p : Sum α β → Prop} : (∀ x, p x) ↔ (∀ a, p (inl a)) ∧ ∀ b, p (inr b) :=
  ⟨fun h => ⟨fun _ => h _, fun _ => h _⟩, fun ⟨h₁, h₂⟩ => Sum.rec h₁ h₂⟩

@[simp] protected theorem «exists» {p : Sum α β → Prop} : (∃ x, p x) ↔ (∃ a, p (inl a)) ∨ ∃ b, p (inr b) :=
```
seems bad to have forall and exists get shadowed, although maybe this isn't an issue since they are keywords anyway?

Reply:
Yeah, probably nicer to have these protected.

Done.